packages/x-components/src/x-modules/tagging/service/__tests__/session.service.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { DefaultSessionService } from '../session.service';
+import { MockedStorageService } from './utils/mock-storage.service';
+
+describe('testing session id service', () => {
+  const prefix = 'test';
+  const mockedStorageService = new MockedStorageService(prefix);
+  const sessionService = new DefaultSessionService(mockedStorageService, 1);
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('creates a new session id', () => {
+    const session = sessionService.getSessionId();
+    expect(mockedStorageService.getPrefix()).toBe(prefix);
+    expect(mockedStorageService.getSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.setSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.removeSessionIdSpy).not.toHaveBeenCalled();
+    expect(mockedStorageService.setSessionIdSpy).toHaveBeenCalledWith(
+      'session-id',
+      expect.any(String),
+      expect.any(Number)
+    );
+    expect(session).toHaveLength(21);
+    expect(session).toMatch(/[A-Za-z0-9_-]/);
+  });
+
+  it('returns an existing session id', () => {
+    mockedStorageService.injectGetResponse('abcd-1234');
+    const session = sessionService.getSessionId();
+    expect(mockedStorageService.getPrefix()).toBe(prefix);
+    expect(mockedStorageService.getSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.setSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.removeSessionIdSpy).not.toHaveBeenCalled();
+    expect(mockedStorageService.setSessionIdSpy).toHaveBeenCalledWith(
+      'session-id',
+      'abcd-1234',
+      expect.any(Number)
+    );
+    expect(session).toBe('abcd-1234');
+  });
+
+  it('removes an existing session id', () => {
+    sessionService.clearSessionId();
+    expect(mockedStorageService.getPrefix()).toBe(prefix);
+    expect(mockedStorageService.getSessionIdSpy).not.toHaveBeenCalled();
+    expect(mockedStorageService.setSessionIdSpy).not.toHaveBeenCalled();
```

Comment:
These checks do not provide any value for this test.
Where and when was the sessionId created? Shouldn't we check after calling `clearSessionId()` that there is no `session` in the `storage`?

Reply:
I'm not sure if we need to test anything related to the `storage`, I don't care what happens in the storage package.

I'll remove this two `not.toHaveBeenCalled` lines.